smart-contracts/contracts/dex-router-osmosis/src/contract.rs

```diff
@@ -0,0 +1,435 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    to_json_binary, BankMsg, Binary, Coin, Deps, DepsMut, Env, Event, MessageInfo, Order, Reply,
+    Response, StdError, StdResult, SubMsg, Uint128,
+};
+use cw2::set_contract_version;
+use mars_owner::OwnerInit::SetInitialOwner;
+use osmosis_std::cosmwasm_to_proto_coins;
+use osmosis_std::types::osmosis::poolmanager::v1beta1::{
+    MsgSwapExactAmountIn, PoolmanagerQuerier, SwapAmountInRoute, TotalPoolLiquidityResponse,
+};
+
+use quasar_types::error::assert_fund_length;
+use std::str::FromStr;
+
+use crate::error::{assert_non_empty_path, ContractError};
+use crate::msg::{BestPathForPairResponse, ExecuteMsg, InstantiateMsg, MigrateMsg, QueryMsg};
+use crate::state::{RecipientInfo, OWNER, PATHS, RECIPIENT_INFO};
+
+const _CONTRACT_NAME: &str = "quasar:dex-router-osmosis";
+const CONTRACT_NAME: &str = "crates.io:dex-router-osmosis";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+const SWAP_REPLY_ID: u64 = 1;
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    _msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+    OWNER.initialize(
```

Comment:
here `OWNER` -(admin) is set on the sender address, which cant be updated afterwards. We should have a `updatOwner` or `updateAfmin` endpoint. @lubkoll @magiodev ?

Reply:
done